openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -538,36 +538,40 @@ def _validate_email(email):
         )
 
 
+def retire_user_comments(user):
+    """
+    Retire the user's discussion comments
+    """
+    try:
+        CCUser.from_django_user(user).retire('Deleted user')
+    except CommentClientRequestError as e:
+        # Ignore error if discussion user does not exist
+        if e.message != u'{"message": "User not found."}':
```

Comment:
While testing this I found this is pretty brittle, but that's really my fault from the `api-integration` implementation, not yours.

```(Pdb) e.message
u'{"message":"User not found."}'
(Pdb) e.message != u'{"message": "User not found."}'
True
```

Can we check `e.status_code != 404` instead?

```
(pdb) e.__dict__
{'status_code': 404, 'message': u'{"message":"User not found."}'}
(pdb) e.status_code != 404
False
```

Reply:
@viadanna :+1:

Reply:
Done